rlclientlib/logger/http_transport_client.h

```diff
@@ -199,6 +207,22 @@ namespace reinforcement_learning {
 
   template <typename TAuthorization>
   int http_transport_client<TAuthorization>::v_send(const buffer& post_data, api_status* status) {
+      return v_send(post_data, _client.get(), status);
+  }
+
+  template <typename TAuthorization>
+  int http_transport_client<TAuthorization>::v_send(const buffer& post_data, unsigned int event_counts, api_status* status) {
+    if (event_counts != 0 &&  _event_counter_status == events_counter_status::ENABLE) {
+      i_http_client* new_client;
```

Comment:
need new http client as every time we need to create new URI based on the events count we get, instead of passing a pointer passing reference now

Reply:
ok well in that case I think this would be a good in-between solution:

- have a member variable of type `unique_ptr` called something like `client_with_event_count`
- have a local raw pointer variable `new_client` to be passed into `create_http_client` and initialized
- call `client_with_event_count.reset(new_client)` which should delete the previous http client and store the freshly 
allocated one in the unique ptr
- pass the unique ptr `client_with_event_count` into `v_send`